platform/android/src/style/sources/geojson_source.cpp

```diff
@@ -66,6 +68,39 @@ namespace android {
         source.as<mbgl::style::GeoJSONSource>()->GeoJSONSource::setGeoJSON(*converted);
     }
 
+    void GeoJSONSource::setGeoJSONStringAsync(jni::JNIEnv& env, jni::String json,
+                                              jni::Object<OnGeoJsonSourceLoadedListener> jListener) {
+        // If another update is running, log an error
+        if (callback) {
+            mbgl::Log::Error(mbgl::Event::JNI, "Error setting GeoJSON: another asynchronous update of this source is being processed.");
+            return;
+        }
+
+        callback = std::make_unique<Actor<FeatureConverter::Callback>>(
+                *Scheduler::GetCurrent(),
+                [this](GeoJSON geoJSON) {
+                    android::UniqueEnv _env = android::AttachEnv();
+
+                    // Update the core source
+                    source.as<mbgl::style::GeoJSONSource>()->GeoJSONSource::setGeoJSON(geoJSON);
```

Comment:
`GeoJSONSource::setGeoJSON` is _itself_ asynchronous, which makes me wonder whether we really need to define this as an asynchronous API at the java level, or if we can just update the behavior of the existing java `setGeoJSON`, which is already implicitly asynchronous. What logic would we expect someone to implement on top of the `onSourceLoaded` callback?